core/src/test/java/com/opentext/ia/sdk/sip/WhenSegmentingDomainObjectsIntoSips.java

```diff
@@ -82,6 +92,109 @@ public void shouldSegmentBySipSize() {
   }
 
   @Test
+  public void shouldSegmentByRandomMaxProspectiveSipSize() throws IOException {
+    int noSips;
+    // Random Strings
+    int sipLimit = randomInt(50, 150);
+    String[] randomArrayofStrings = randomArrayOfStrings(randomInt(4, 8));
+
+    int domainObjectSize = 0;
+    for (String thisOne : randomArrayofStrings) {
+      domainObjectSize += thisOne.length();
+    }
+    if (domainObjectSize > sipLimit) {
+      expectedException.expect(DomainObjectTooBigException.class);
+    }
+    noSips = executeSegmentByProspectiveSipSize(sipLimit, randomArrayofStrings);
+    assertEquals("SIP counts - Random generated Strings", expected, noSips);
+  }
+
+  @Test
+  public void shouldSegmentByMaxProspectiveSipSize() throws IOException {
+    int noSips;
+    // Boundary Test - 45 size and 90 limit
+    noSips = executeSegmentByProspectiveSipSize(90,
+        new String[] { "Hello", "Doman", "yuiopqwertyuiop", "poiuytrewqpoiuytrewq" });
+    assertEquals("SIP counts - Fixed Length Strings", expected, noSips);
+
+    // Boundary Test - exact match
```

Comment:
Split this off into a separate test, so that we know whether the seconds part works or not when the first part fails.